firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteLruReferenceDelegate.java

```diff
@@ -70,8 +70,14 @@ public LruGarbageCollector getGarbageCollector() {
   }
 
   @Override
-  public long getTargetCount() {
-    return persistence.getQueryCache().getTargetCount();
+  public long getSequenceNumberCount() {
+    long targetCount = persistence.getQueryCache().getTargetCount();
+    long orphanedDocumentCount =
+        persistence
+            .query(
+                "SELECT COUNT(*) FROM (SELECT sequence_number FROM target_documents GROUP BY path HAVING COUNT(*) = 1 AND target_id = 0)")
```

Comment:
I don't know SQLite, and I assume you tried "SELECT COUNT(*) FROM target_documents GROUP BY path HAVING COUNT(*) = 1 AND target_id = 0"?

Reply:
I did indeed try that first. I believe `COUNT(*)` in the projection interacts poorly with `GROUP BY` in terms of getting what we want here. Using the subquery technique works, and in retrospect is I think a little more obvious when compared with the similar query used to find the sequence numbers.